[admin] Add criteria for distributions (#376)
open-telemetry/opentelemetry-collector-releases: distributions/README.md

# Criteria for Supported Distributions

Support for a distribution implies all the duties covered by the Approver and Maintainer role requirements. In addition, support means that the Collector SIG is the owner and maintainer of the binaries/images of the different Collector distributions and is responsible for the pipeline that produces those artifacts.

Distributions supported by the Collector SIG should fulfill the following criteria:

> 1. Support at least one distribution that is recommended for production which includes support for Prometheus, Jaeger, Zipkin, and OpenCensus.

Reviewer: This one is a bit unclear. It doesn't seem like a criterion for distributions but rather a requirement for the Collector SIG

Author: That's fair, I could move it out of the list and just state it as a thing we must remember to do.

Author: Moved

# Criteria for Supported Distributions

Support for a distribution implies all the duties covered by the Approver and Maintainer role requirements. In addition, support means that the Collector SIG is the owner and maintainer of the binaries/images of the different Collector distributions and is responsible for the pipeline that produces those artifacts.

> The Collector SIG must support at least one distribution that is recommended for production which includes support for Prometheus, Jaeger, Zipkin, and OpenCensus.

Reviewer 2: ```suggestion
The Collector SIG will support at least one distribution that is recommended for production which includes support for Prometheus, Jaeger, Zipkin, and OpenCensus.
```
Nit: I think this reads nicer when written as a commitment.

Distributions supported by the Collector SIG should fulfill the following criteria:

> 1. Serve a specific purpose and those purposes should have minimal overlap.

Reviewer 2: ```suggestion
1. Serve a specific purpose that has minimal overlap with the purpose of any other distribution.
```

2. Meet general needs and not be too niche.

> 3. Only target the needs of the OpenTelemetry project.

Reviewer 3: These two goals seem to be in conflict.  Do we have guidance on how to resolve the tension?  Goal 2) is also ill-defined; how am I to know how niche is "too niche"?  That's a judgement that will vary wildly based on perspective.

Author: The intention behind 2 is to stop us from supporting distributions that won't get a lot of use.  For example, if someone came to us and wanted us to support a distribution that targeted their specific environment needs, or if someone came and said "I think the community should support am image with only components X, Y and Z bc I need it", we could point to this rule.

The intention behind 3 is to prevent us from supporting distributions for other third-parties. The issue was raised specifically around venders. We don't intend to be the only source of collector distributions - we do intent to be a source of collector distributions that fit our needs.

When stated verbosely I don't feel they are in conflict, but I'm willing to continue wordsmithing until the intent is clearer.

How does this sound:

```
  2. Should meet general needs and be desired by many users
  3. Should not be specific to any vendor
```

@open-telemetry/collector-contrib-approvers please review.

Reviewer 4: I like that wording more than the current one, I also [struggled to understand (3) at first](https://github.com/open-telemetry/opentelemetry-collector-releases/pull/376#discussion_r1275234338), this sounds clearer

Reviewer 3: That sounds like a better formulation, thank you.

> 4. May be focused on development and proof of concept use cases.  The distribution should clearly indicate whether the Collector SIG recommends the distribution be used in Production environments.

Reviewer 2: ```suggestion
4. May be focused on development or proof of concept use cases.  The distribution should clearly indicate whether the Collector SIG recommends the distribution be used in production environments.
```
nit

5. Must only include components from the `opentelemetry-collector` and `opentelemetry-collector-contrib` repositories.

> 6. Have a clearly defined list of criteria for which components are included.

Reviewer 5: Before we merge, can we have tracking issues to ensure we define these lists for our existing distributions?

Author: @djaglowski @evan-bradley we're currently using https://github.com/open-telemetry/opentelemetry-collector-releases/issues/360 to track the criteria building.  The discussion for Core continues but contrib has stalled, so once this is merged I plan to do a PR proposing Contrib's criteria.

> 7. Must include the following assets except where the specific purpose of the distribution is naturally associated with a subset of these assets. In such cases, it should be clearly stated which assets are skipped and why.  Additional asset may be included if the distro desires:

Reviewer 5: > Additional asset may be included if the distro desires

Who decides what is desirable? Maybe it should say something along the lines of "if the maintainers agree"

Author: Good idea.

Author: Updated

> 7. Must include the following assets except where the specific purpose of the distribution is naturally associated with a subset of these assets. In such cases, it should be clearly stated which assets are skipped and why.  Additional asset may be included if the distro desires:

Reviewer 2: ```suggestion
7. Must include the following assets except where the specific purpose of the distribution is naturally associated with a subset of these assets. In such cases, it should be clearly stated which assets are skipped and why.  Additional assets may be included if the distro desires:
```

Support for a distribution implies all the duties covered by the Approver and Maintainer role requirements. In addition, support means that the Collector SIG is the owner and maintainer of the binaries/images of the different Collector distributions and is responsible for the pipeline that produces those artifacts.

> The Collector SIG must support at least one distribution that is recommended for production which includes support for Prometheus, Jaeger, Zipkin, and OpenCensus.

Reviewer: Jaeger and OpenCensus will be deprecated and removed at some point. Will we update this one later or better rephrase it now?

Author: I say we update it once we aren't being held to that expectation.  As far as I am aware we still are. @jpkrohling can you confirm?

Reviewer 6: They will, but they are not yet. I'd keep them here, and remove when time comes.
